scripts/perftests/PerformanceTestUtilities.ps1

```diff
@@ -0,0 +1,214 @@
+    function DownloadNuGetExe([string]$version, [string]$downloadDirectory)
+    {        
+        $NuGetExeUriRoot = "https://dist.nuget.org/win-x86-commandline/v"
+        $NuGetExeSuffix = "/nuget.exe"
+
+        $url = $NuGetExeUriRoot + $version + $NuGetExeSuffix
+        $Path =  $downloadDirectory + "\" + $version
+        $ExePath = $Path + "\nuget.exe"
+
+        if (!(Test-Path($ExePath)))
+        {
+            Log "Downloading $url to $ExePath" -color "Green"
+            New-Item -ItemType Directory -Force -Path $Path > $null
+            Invoke-WebRequest -Uri $url -OutFile $ExePath
+        }
+        return GetAbsolutePath $ExePath
+    }
+
+    
+    # The format of the URL is assumed to be https://github.com/NuGet/NuGet.Client.git. The result would be NuGet-Client-git
+    function GenerateNameFromGitUrl([string]$gitUrl)
+    {
+        return $gitUrl.Substring($($gitUrl.LastIndexOf('/') + 1)).Replace('.','-')
+    }
+
+    # Appends the log time in front of the log statement with the color specified. 
+    function Log([string]$logStatement, [string]$color)
+    {
+        if(-not ([string]::IsNullOrEmpty($color)))
+        {
+            Write-Host "$($(Get-Date).ToString()): $logStatement" -ForegroundColor $color
+        }
+        else
+        { 
+            Write-Host "$($(Get-Date).ToString()): $logStatement"
+        }
+    }
+
+    # Given a relative path, gets the absolute path from the current directory
+    function GetAbsolutePath([string]$Path)
+    {
+        $Path = [System.IO.Path]::Combine(((pwd).Path), ($Path));
+        $Path = [System.IO.Path]::GetFullPath($Path);
+        return $Path;
+    }
+
+    # Writes the content to the given path. Creates the folder structure if needed
+    function OutFileWithCreateFolders([string]$path, [string]$content){
+        $folder = [System.IO.Path]::GetDirectoryName($path)
+        If(!(Test-Path $folder))
+        {
+            & New-Item -ItemType Directory -Force -Path $folder > $null
+        }
+        Add-Content -Path $path -Value $content
+    }
+
+    # Gets a list of all the nupkgs recursively in the global packages folder
+    function GetAllPackagesInGlobalPackagesFolder([string]$packagesFolder)
+    {
+        if(Test-Path $packagesFolder)
+        {
+            $packages = Get-ChildItem $packagesFolder\*.nupkg -Recurse
+            return $packages
+        }
+        return $null
+    }
+
+    # Gets a list of all the files resursively in the given folder
+    function GetFiles([string]$folder)
+    {
+        if(Test-Path $folder)
+        {
+            $files = Get-ChildItem $folder -recurse
+            return $files
+        }
+        return $null
+    }
+
+    # Determines if the client is dotnet.exe by checking the path.
+    function GetClientName([string]$nugetClient)
+    {
+        return $nugetClient.Substring($($nugetClient.LastIndexOf([System.IO.Path]::DirectorySeparatorChar) + 1))
```

Comment:
Can't you simplify this to?

    return [System.IO.Path]::GetFileName($nugetClient)